packages/cache/src/internal/cacheHttpClient.ts

```diff
@@ -73,13 +73,18 @@ function createHttpClient(): HttpClient {
 
 export function getCacheVersion(
   paths: string[],
-  compressionMethod?: CompressionMethod
+  compressionMethod?: CompressionMethod,
+  crossOsEnabled = false
 ): string {
-  const components = paths.concat(
-    !compressionMethod || compressionMethod === CompressionMethod.Gzip
-      ? []
-      : [compressionMethod]
-  )
+  const components = paths
+    .concat(
+      !compressionMethod || compressionMethod === CompressionMethod.Gzip
+        ? []
+        : [compressionMethod]
+    )
+    .concat(
+      process.platform !== 'win32' || crossOsEnabled ? [] : ['windows-only']
```

Comment:
Don't we need to handle Windows -> linux case as well? Would it be better to name this `non-cross-os`?

Reply:
Actually this will get handled automatically as windows will create cache with this suffix version